### PR TITLE
commands: introduce non_desktop command for output

### LINKED ISSUE #7021: Create some way of dynamically setting displays as desktop or non-desktop
Two use-cases:
* I have a display that says it's non-desktop (like say an Index) and for some reason I want to extend the desktop to it anyway
* I have a display that looks to Sway like a normal monitor (like say a Project North Star AR headset) and I need to tell Sway to treat it as non-desktop so that Monado can do direct mode with it

### PR DESCRIPTION
**Pull request by bl4ckb0ne**
This command allows users to set a given output as non-desktop. The output is offered to the drm-lease-manager interface if available, for clients to lease.

Closes #7021

### CODE REVIEW

File: sway/commands/output/non_desktop.c [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+	struct output_config *oc = config->handler_context.output_config;
+
+	if (strcmp(oc->name, "*") == 0) {
+		return cmd_results_new(CMD_INVALID,
+			"Cannot apply non_desktop to all outputs");
+	}
+
+	if (parse_boolean(argv[0], true)) {
+		int index = list_seq_find(root->outputs, find_output, oc->name);
```
Comment (emersion) [inline sway/commands/output/non_desktop.c:37]: We already have a function to do this:

```
struct sway_output *sway_output = all_output_by_name_or_id(oc->name);
```

It also handles output identifiers.

File: sway/commands/output/non_desktop.c [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		if (index == -1) {
+			return cmd_results_new(CMD_FAILURE, "unknown output %s ", oc->name);
+		}
+
+		struct sway_output *output = root->outputs->items[index];
+		struct wlr_output *wlr_output = output->wlr_output;
+		wlr_output->non_desktop = true;
```
Comment (emersion) [inline sway/commands/output/non_desktop.c:44]: Hmm. Do we really need to do this? I don't like writing to the `wlr_output` struct.

File: sway/commands/output/non_desktop.c [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+			wlr_output);
+
+		output_disable(output);
+		wlr_output_layout_remove(root->output_layout, wlr_output);
+
+		struct sway_output_non_desktop *non_desktop =
+			output_non_desktop_create(wlr_output);
+		list_add(root->non_desktop_outputs, non_desktop);
+		
+		if (server.drm_lease_manager) {
```
Comment (emersion) [inline sway/commands/output/non_desktop.c:56]: Maybe fail early in this case?

**Comment by emersion** [comment]
I'd prefer to name this something else. "non desktop" is just a hint coming from the kernel. This command allows one to lease "desktop" outputs.

tl;dr something like "drm_lease" would be a better name IMHO.

File: sway/sway-output.5.scd [hunk 4]
```diff
@@ -180,6 +180,11 @@ must be separated by one space. For example:
 	updated to work with different bit depths. This command is experimental,
 	and may be removed or changed in the future.
 
+*output* <name> non_desktop on|off
+	Enables or disables the non-desktop property of a given output. When
+	enabled, the output is disabled and offered to the drm-lease-manager
+	interface if it is available.
```
Comment (emersion) [inline sway/sway-output.5.scd:186]: Maybe we should leave this command undocumented for a bit and see how it goes… A bit like `create_output` and `output XXX unplug`. If we do document it, then we're stuck with it ~forever.

File: sway/commands/output/non_desktop.c (hunk 1 above)
Comment (bl4ckb0ne) [reply sway/commands/output/non_desktop.c:37]: `all_output_by_name_or_id` searches in `root->all_outputs`,  is it better? I didnt get the difference between the two lists.

File: sway/commands/output/non_desktop.c (hunk 2 above)
Comment (bl4ckb0ne) [reply sway/commands/output/non_desktop.c:44]: I don't recally why I did that, maybe a leftover from a previous draft. I'll remove it.

**Comment by bl4ckb0ne** [comment]
Now that you're saying it out loud it does make a lot of sense to have a `lease` command instead of using `non_desktop`. But that changes the scope of the whole patchset. 

Should true non-desktop output still be treated differently? Or should we treat them as regular output and already lease them unless the users requests to withdraw them?

**Comment by bl4ckb0ne** [comment]
Closed in favour of #7207
